content/en/docs/reference/measuring-kubernetes/scalability-thresholds.md

```diff
@@ -0,0 +1,83 @@
+date: 2019-06-28
+full_link: /docs/reference/instrumentation/scalability-thresholds/
+short_description: >
+  This document describes the thresholds that must be adhered to for SLOs to be satisfied.
+---
+
+## Background
+
+As described in [How we define scalability] document, it is impossible
+to provide guarantees in a generic situation. One of the prerequisites
+for SLOs being satisfied is keeping the load in the cluster within
+recommended limits. This document is trying to explicitly summarize
+dimensions and limits themselves.
+
+[How we define scalability](/docs/reference/measuring-kubernetes/slos/)
+
+## Kubernetes thresholds
+
+Scalability dimensions and thresholds are very complex topic. In fact,
+configurations that Kubernetes supports create `Scalability Envelope`:
+
+{{< figure src="/docs/images/scalability-envelope.png" alt="scalability-envelope-diagram" class="scalability-envelope-large" caption="Figure. Scalability Envelope" link="https://github.com/kubernetes/community/blob/master/sig-scalability/configs-and-limits/scalability-envelope.png?raw=true" >}}
+
+Some the properties of the envelope:
+1. It's NOT a cube, because dimensions are sometimes not independent.
+1. It's NOT convex.
+1. As you move farther along one dimension, your cross-section wrt other
+   dimensions gets smaller.
+1. It's bounded.
+1. It's decomposable into smaller envelopes.
+
+You can learn more about it in this [KubeCon talk] (or [KubeCon slides]).
+
+There are couple caveats to the thresholds we are presenting below:
+1. In majority of cases, thresholds are NOT hard limits - crossing
+   the limit results in degraded performance and doesn't mean cluster
+   immediately fails over.
+1. **Many of the thresholds (for cluster scope) are given for the largest
+   possible cluster. For smaller clusters, the limits are proportionally
+   lower.**
+1. The thresholds may differ (hopefully be non-decreasing) across Kubernetes
+   releases. The threshold below are given for Kubernetes head. <br/>
+   **TODO:** We are planning to start versioning the table below, but we
+   are not there yet.
+1. Given that configuration influences thresholds, we are assuming vanilla
+   Kubernetes setup.
+
+The table below is **NOT exhaustive** - more content is coming soon.
+
+| Quantity                    | Threshold scope=namespace | Threshold: scope=cluster |
+|-----------------------------|---------------------------|--------------------------|
+| #Nodes                      | n/a                       | 5000                     |
+| #Namespaces                 | n/a                       | 10000                    |
+| #Pods                       | 3000                      | 150000                   |
+| #Pods per node              | min(110, 10*#cores)       | min(110, 10*#cores)      |
+| #Services                   | 5000                      | 10000                    |
+| #All service endpoints      | TBD                       | TBD                      |
```

Comment:
I think we should either fill those TBDs or get rid of them from here.

TBDs make sense in the community page, but here (given the comment above about the list not being exhaustive), we should probably remove those lines.